### ISSUE TITLE
Refactor AbstractContainerPanel and WinePrefixContainerPanel

### ISSUE THREAD

**Issue by madoar** [PhoenicisOrg/POL-POM-5#819]
Currently the class `AbstractContainerPanel` and its subclass `WinePrefixContainerPanel`, which is opened when a user selects a container in the "Containers" tab, need to be refactored:

- when opening a `WinePrefixContainerPanel` it takes too much space. If you don't resize your POL 5 frame, this leads to the `WinePrefixContainerPanel` being larger than the space it's shown in, which leads to the hiding of the close button in the default size
- in addition the `AbstractContainerPanel` contains no title, that is shown in the header row together with the close button. Here we should show the name of the selected container
- the consumers inside `WinePrefixContainerPanel` are currently automatically initialized, which takes a lot of space and looks quite ugly with the automatic line breaks they contain. Here I would suggest to not automatically initialize them, or, in case we absolutely want to prevent `null` consumers, we should use `Optional<Consumer<WinePrefixContainerDTO>>`, which can then be used with `consumer.ifPresent(c -> c.accept(winePrefixContainerDTO))`
- `WinePrefixContainerPanel` still uses the ThemeManager to set icons, which makes it hard to switch the icons with a theme change

**Comment by plata** [comment]
I'm not sure it's a good idea to have all this in one issue. In fact those are distinct topics which can/should be handled individually.

**Comment by madoar** [comment]
Fell free to split it up when working on this :)

**Comment by plata** [comment]
I only would like to avoid that people do not grab a task because the whole thing seems to be too huge.

**Comment by madoar** [comment]
I'm not sure how other people are handling refactoring tasks, but when I refactor a class or a functionality, I normally try to fix all issues I know at once, because every normal (quick) fix normally breaks the "symmetry" of the existing code and looks like someone used some glue to paste a small piece to the code. If you do this too often it's bad.
In addition a lot of these things quite small changes, like setting the title which is one line, the biggest change will be #821.

**Comment by madoar** [comment]
We may have another problem with this longterm. If we want to implement engines in a plugable way in the script repositories, then we need to decide how to offer engine specific container functionality.
Currently this is hardcoded inside `WinePrefixContainerPanel`, can we change this somehow?

**Comment by plata** [comment]
We have to move all of this to JS. Without thinking a lot about it, we could do something like:
* Have a generic function `getEngineTools` in the JS engine which returns a list of function names and tool button icons. If you click a button, we call the provided function from Java.
* "Display" and "Input" are moved to a generic "Settings" tab (so we have "Information", "Settings", "Tools").
* Settings must be provided by JS, e.g. `getSettings` function which returns a list of `EngineSetting`. `EngineSetting` has a name, type (string, number, choice) etc.

**Comment by madoar** [comment]
If we do this, wouldn't it make the creation of a new engine type much more complicated and work intensive? Let's say we want to create a new engine how many methods and functionality would you need to implement until you have something you can test in POL 5?

**Comment by plata** [comment]
We can have all of this optional. If you don't want to provide any tool, you do not have to implement anything. To do this, we derive all engine types from a base class `Engine` which e.g. returns an empty list for `getEngineTools`.

**Comment by plata** [comment]
I will move this information to #620.

**Comment by madoar** [comment]
Good idea

**Comment by madoar** [comment]
I'm not sure if we should work on this issue currently or if we should do this together with or after #620.
The reason for this is, that if we do this first, we'll need to throw a lot of the changes done out of the window when we work on #620. 

What should be done in my opinion at this point in time is #821.

**Comment by plata** [comment]
split into:
- #1000 
- #1001 
- #1002 
- #1003
